src/canary.rs

```diff
@@ -185,6 +185,15 @@ impl Canary {
     }
 }
 
+fn round_up(n: u32, k: u32) -> u32 {
+    let rem = n % k;
+    if rem == 0 {
+        n
+    } else {
+        n - rem + 4
```

Comment:
```suggestion
        n + 4 - rem
```

This will resolve the ambiguity between `(n - rem) + 4` and `n - (rem + 4)`.